modules/va_facilities/app/controllers/va_facilities/v1/nearby_controller.rb

```diff
@@ -70,21 +76,34 @@ def validate_params
         validate_type_and_services_known unless params[:type].nil?
       end
 
-      def get_query_method(params)
+      def get_lat_lng(params)
         obs_fields = params.keys.map(&:to_sym)
         location_type = REQUIRED_PARAMS.find do |loc_type, req_field_names|
```

Comment:
When does this block make the decision to return a location type? It seems like it only even returns the last location type, and only if all the params are present. Otherwise, it's nil. Am I misunderstanding?

Reply:
it iterates over the`REQUIRED_PARAMS` and returns the first one that has no missing fields.

Reply:
In that case,  I think the logic here is overly complicated. The `break` is unnecessary since `find` just needs a condition to return the thing. Here's what we could do:
```suggestion
        location_type = REQUIRED_PARAMS.keys.find do |loc_type|
           (REQUIRED_PARAMS[loc_type] - obs_fields).empty?
        end
```

Reply:
That will return an array of key, value. So for example, if the the params are `{lat: 45, lng: 45}` then that function will return `[:lat_lng, [:lat, :lng]]`

this was also discussed here:
https://github.com/department-of-veterans-affairs/vets-api/pull/3273/files/a8c59804eca6e24421bd527e208382c180665c6f

Reply:
Actually, in my code, we are iterating over an array of the keys, not the hash. So here we are iterating over `[:address, :lat_lng]`. Make sense?

Reply:
ah cool, didn't see the keys part